pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -254,16 +254,21 @@ func (r *Reconciler) PushSecretToProviders(ctx context.Context, stores map[esapi
 		if err != nil {
 			return out, fmt.Errorf("could not get secrets client for store %v: %w", store.GetName(), err)
 		}
-		for _, ref := range ps.Spec.Data {
-			secretValue, ok := secret.Data[ref.Match.SecretKey]
+		for _, data := range ps.Spec.Data {
+			secretValue, ok := secret.Data[data.Match.SecretKey]
 			if !ok {
-				return out, fmt.Errorf("secret key %v does not exist", ref.Match.SecretKey)
+				return out, fmt.Errorf("secret key %v does not exist", data.Match.SecretKey)
 			}
-			err := client.PushSecret(ctx, secretValue, ref.Match.RemoteRef)
+
+			if data.Metadata != nil && data.Match.RemoteRef.Property != "" {
+				return out, fmt.Errorf("cannot specify metadata and property at the same time")
```

Comment:
I spent a lot of time thinking about how to support the property and the metadata at the same time. This is quite challenging so I just gave it up for now. 

For example, if we use Kubernetes provider, it is not clear how we should deal with the metadata attached to each secret. To avoid those tough questions, I just decided to focus on PushSecret without the property field.

Even though I validate the property field in the controller, it might be better if we leave it to each provider, but WDYT?

Reply:
Right, this is a tough one. For a single `PushSecret` we can set the metadata, when multiple PushSecret write multiple properties to the same secret there might be a conflict metadata-wise.

Agree, we should leave it up to the provider.